hs_core/views/resource_folder_hierarchy.py

```diff
@@ -167,11 +167,12 @@ def data_store_structure(request):
                 # accept any extension
                 main_extension = ""
 
-            _ , file_extension = os.path.splitext(fname)
-            if file_extension and main_extension.endswith(file_extension):
+            _, file_extension = os.path.splitext(fname)
+            if file_extension and main_extension.endswith(file_extension) and main_extension != ".csv":
```

Comment:
This block of code is related to how we decide based on the main file of an aggregation to show that aggregation as virtual folder in UI. For example, netcdf, raster, geofeatue aggregations are displayed as virtual folder. For a single file aggregation there is no main file with any specific extension, so it never becomes part of the virtual folder (`main_extension.endswith(file_extension)` condition fails). So this special exception for csv is to avoid displaying csv aggregation as a virtual folder.